app/views/user_credentials/index.html.erb

```diff
@@ -39,6 +39,7 @@
       <div class="btn-group">
         <%= link_to new_user_credential_path, class: "btn btn-default" do %><span class="glyphicon glyphicon-plus"></span> New Credential<% end %>
         <%= link_to user_credentials_path(format: :json), class: "btn btn-default" do %><span class="glyphicon glyphicon-cloud-download"></span> Download Credentials<% end %>
+        <%= form_tag import_user_credentials_path, multipart: true, class: "btn btn-default" do %><span class="glyphicon glyphicon-upload"><label class="btn btn-primary" for="file"><%= file_field_tag :file %></span><%= submit_tag "Upload File"%></label><% end %>
```

Comment:
By the way, it looks like you have <span><label></span></label> which isn't valid HTML.

But, maybe we could do this?

```html
      <div class="btn-group">
        <%= link_to new_user_credential_path, class: "btn btn-default" do %><span class="glyphicon glyphicon-plus"></span> New Credential<% end %>
        <%= link_to user_credentials_path(format: :json), class: "btn btn-default" do %><span class="glyphicon glyphicon-cloud-download"></span> Download Credentials<% end %>
        <%= link_to '#', data: { toggle: 'modal', target: '#credentials-upload' }, class: "btn btn-default credentials-upload-button" do %><span class="glyphicon glyphicon-upload"></span> Upload Credentials<% end %>
      </div>

      <div id="credentials-upload" class="modal fade" tabindex="-1" role="dialog">
        <div class="modal-dialog modal-sm">
          <div class="modal-content">
            <div class="modal-header">
              <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
              <h4 class="modal-title">Upload Credentials</h4>
            </div>
            <div class="modal-body">
              <p>Upload a credentials file that you have previously exported from a Huginn instance.</p>
            </div>
            <div class="modal-footer">
              <%= form_tag import_user_credentials_path, multipart: true do %>
                <%= file_field_tag :file %>
                <%= button_tag 'Cancel', class: 'btn btn-default', 'data-dismiss' => 'modal' %>
                <%= submit_tag 'Upload', class: 'btn btn-primary' %>
              <% end %>
            </div>
          </div>
        </div>
      </div>
```